docs/standard/base-types/regular-expression-source-generators.md

```diff
@@ -0,0 +1,400 @@
+
+To help with these issues, .NET provides a method <xref:System.Text.RegularExpressions.Regex.CompileToAssembly%2A?displayProperty=nameWithType>. This method enables the same IL that would have been generated for `RegexOptions.Compiled` to instead be written to a generated assembly on disk, and that assembly can then be referenced as a library from your app. This has the benefit of avoiding the startup overheads involved in parsing, optimizing, and outputting the IL for the expression, as that can all be done ahead of time rather than each time the app is invoked. Further, that assembly could be ahead-of-time compiled with technology like ngen or crossgen, avoiding most of the associated JIT costs as well.
+
+`Regex.CompileToAssembly` itself has problems, however. First, it's not user-friendly. Because a utility was required to call `CompileToAssembly` to produce an assembly your app would reference, there's relatively little use for this otherwise valuable feature. On .NET [Core], `CompileToAssembly` has never been supported, as it requires the ability to save reflection-emit code to assemblies on disk, which also isn't supported. This is where source generation becomes valuable.
+
+## Source generation
+
+.NET 7 introduces a new `RegexGenerator` source generator. When the C# compiler was rewritten as the ["Roslyn" C# compiler](../../csharp/roslyn-sdk/index.md), it exposed object models for the entire compilation pipeline, as well as analyzers. More recently, Roslyn enabled source generators. Just like an analyzer, a source generator is a component that plugs into the compiler and is handed all of the same information as an analyzer, but in addition to being able to emit diagnostics, it can also augment the compilation unit with additional source code. The .NET 7 SDK includes a new source generator that recognizes the new <xref:System.Text.RegularExpressions.GeneratedRegexAttribute> on a partial method that returns `Regex`. The source generator provides an implementation of that method that implements all the logic for the `Regex`. For example, you might have written code like this:
+
+```csharp
+private static readonly Regex s_abcOrDefGeneratedRegex =
+    new(pattern: "abc|def",
+        options: RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+private static void EvaluateText(string text)
+{
+    if (s_abcOrDefGeneratedRegex.IsMatch(text))
+    {
+        // Take action with matching text
+    }
+}
+```
+
+You can now rewrite the previous code as follows:
+
+```csharp
+private static readonly Regex s_abcOrDefGeneratedRegex = AbcOrDefGeneratedRegex();
+
+[GeneratedRegex("abc|def", RegexOptions.IgnoreCase | RegexOptions.Compiled, "en-US")]
+private static partial Regex AbcOrDefGeneratedRegex();
```

Comment:
There's no benefit to caching this into a static field.  The source generator does that itself.  If you look at the generated code for this, you'll see the generated AbcOrDefGeneratedRegex method is essentially just `AbcOrDefGeneratedRegex() => s_cachedField;`.

Reply:
I thought the purpose of this section was to show exactly what the analyzer with the applied fix will do, I can adjust this. The analyzer's applied fix though doesn't behave this way, instead, it will re-write your code like the resulting code above.

Reply:
The analyzer isn't mentioned in this section.  The description of this code sample states: "You can now rewrite the previous code as follows:"